core/platform/auth/firebase_auth_services.py

```
@@ -184,6 +185,102 @@ def get_all_external_accounts() -> List[auth_domain.ExternalAccount]:
+    Raises:
+        ValueError. If any of the acconts have a malformed ID.
+        auth_domain.AuthProviderError. If an error occurs during the operation.
+    """
+    uids = (acc.auth_id for acc in accounts if acc.auth_id)
+    try:
+        while batch := list(itertools.islice(uids, feconf.FIREBASE_BATCH_SIZE)):
+            _log_delete_users_result(
+                firebase_auth.delete_users(batch, force_delete=True)
+            )
+    except firebase_exceptions.FirebaseError as e:
+        raise auth_domain.AuthProviderError('Failed to delete accounts') from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)